_comments-app/components/TextEditor.js

```diff
@@ -0,0 +1,253 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Editor, EditorState, RichUtils} from 'draft-js';
+
+// Custom overrides for inline code style.
+const styleMap = {
+  CODE: {
+    backgroundColor: '#fff',
+    border: '1px solid #ddd',
+    borderRadius: '.2em',
+    color: '#9e023b',
+    fontFamily: 'Menlo, Monaco, "Andale Mono", "lucida console", "Courier New", monospace',
+    fontSize: '.9em',
+    lineHeight: '1.5em',
+    margin: '-1px 0',
+    padding: '.15em .3em',
+  },
+};
+
+/**
+ * Set up classes for block style elements.
+ *
+ * @param block
+ * @returns {*}
+ */
+function getBlockStyle(block) {
+  switch (block.getType()) {
+  case 'blockquote':
+    return 'text-editor__editor__blockquote';
+  default:
+    return null;
+  }
+}
+
+/**
+ * The individual editor buttons that toggle text styles.
+ */
+class StyleButton extends React.Component {
+  constructor() {
+    super();
+    this.onToggle = this.onToggle.bind(this);
+  }
+  onToggle(e) {
+    e.preventDefault();
+    this.props.onToggle(this.props.style);
+  }
+  render() {
+    let className = 'text-editor__style-button';
+    if (this.props.active) {
+      className += ' text-editor__style-button--active';
+    }
+
+    return (
+      <span className={className} onMouseDown={this.onToggle}>
+        {this.props.label}
+      </span>
+    );
+  }
+}
+
+StyleButton.propTypes = {
+  onToggle: PropTypes.func.isRequired,
+  style: PropTypes.string.isRequired,
+  active: PropTypes.bool.isRequired,
+  label: PropTypes.string.isRequired,
+};
+
+/**
+ * Define block-level style buttons.
+ */
+const BLOCK_TYPES = [
+  {label: 'Heading', style: 'header-three'},
```

Comment:
I'm wondering if we could remove either heading or subheading. My sense that commenters would most likely need only a single heading level.

Reply:
I think that's fine, I'll remove it. We can easily add it back in later if we want to.